src/main/scala/org/apache/mesos/chronos/scheduler/mesos/MesosDriverFactory.scala

```diff
@@ -45,4 +57,31 @@ class MesosDriverFactory(val mesosScheduler: Scheduler, val frameworkInfo: Frame
     mesosDriver.get.stop(true)
     mesosDriver = None
   }
+
+
+  /**
+   * Create the optional credentials instance, used to authenticate calls from Chronos to Mesos.
+   */
+  def buildMesosCredentials(principal: String, secretFile: Option[String]): Credential = {
+
+    val credentialBuilder = Credential.newBuilder()
+      .setPrincipal(principal)
+
+    secretFile foreach { file =>
+      try {
+        val secretBytes = ByteString.readFrom(new FileInputStream(file))
+        if ((new File(file)).canWrite())
```

Comment:
More importantly, the file should not be world-readable. I'd rather the current user be able to change their credential file than a stranger be able to read and use it. In rwx modes, this would be 400 (-r--------) or maybe 440 (-r--r-----) if they want it group-readable.